pyopenms_viz/_core.py

```diff
@@ -748,103 +792,124 @@ def _computed_num_bins(self):
 
     def plot(self):
         """Standard spectrum plot with m/z on x-axis, intensity on y-axis and optional mirror spectrum."""
-
-        # Prepare data
-        spectrum = self._prepare_data(self.data)
-        if self.reference_spectrum is not None:
-            reference_spectrum = self._prepare_data(self.reference_spectrum)
+        
+        # Check if tiling is requested
+        tile_by = self._config.tile_by if hasattr(self._config, "tile_by") else None
+        if tile_by and tile_by in self.data.columns:
+            # Group data by tile_by column
+            grouped = self.data.groupby(tile_by)
+            num_groups = len(grouped)
+            from math import ceil
+            tile_columns = self._config.tile_columns if hasattr(self._config, "tile_columns") else 1
+            tile_rows = ceil(num_groups / tile_columns)
+            figsize = self._config.tile_figsize if hasattr(self._config, "tile_figsize") else (10, 15)
+            fig, axes = plt.subplots(tile_rows, tile_columns, figsize=figsize, squeeze=False)
+            axes = axes.flatten()
```

Comment:
matplotlib specific code should not be here as this is abstracted. Add this functionality to the create_figure method instead

https://github.com/OpenMS/pyopenms_viz/blob/242c9505c3efa2834f35af5031ff6c58d564e381/pyopenms_viz/_matplotlib/core.py#L63C9-L63C30